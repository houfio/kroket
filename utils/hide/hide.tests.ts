import { act, renderHook } from '@testing-library/react-hooks';

import { useHide } from '.';

it('should hide all children', () => {
  const { result: { current: [hide] } } = renderHook(() => useHide());

  document.body.innerHTML = `
    <main>
      <button>
        test
      </button>
    </main>
    <div>
      test
    </div>
  `;

  act(() => {
    hide(document.body);
  });

  expect(document.body).toMatchSnapshot();
});

it('should hide all children except the excluded', () => {
  const { result: { current: [hide] } } = renderHook(() => useHide());

  document.body.innerHTML = `
    <main>
      <button>
        test
      </button>
    </main>
    <div>
      test
    </div>
  `;

  act(() => {
    hide(document.body, 'div');
  });

  expect(document.body).toMatchSnapshot();
});

it('should unhide all hidden children', () => {
  const { result: { current: [hide, unhide] } } = renderHook(() => useHide());

  document.body.innerHTML = `
    <main>
      <button>
        test
      </button>
    </main>
    <div>
      test
    </div>
  `;

  act(() => {
    hide(document.body);
    unhide();
  });

  expect(document.body).toMatchSnapshot();
});

it('shouldn\'t hide when there are hidden children', () => {
  const { result: { current: [hide] } } = renderHook(() => useHide());

  document.body.innerHTML = `
    <main>
      <button>
        test
      </button>
    </main>
    <div>
      test
    </div>
  `;

  act(() => {
    hide(document.body);
    hide(document.body.getElementsByTagName('main')[0]);
  });

  expect(document.body).toMatchSnapshot();
});
