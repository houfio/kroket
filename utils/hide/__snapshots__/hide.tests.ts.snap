// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide all children 1`] = `
<body>
  
    
  <main
    aria-hidden="true"
  >
    
      
    <button>
      
        test
      
    </button>
    
    
  </main>
  
    
  <div
    aria-hidden="true"
  >
    
      test
    
  </div>
  
  
</body>
`;

exports[`should hide all children except the excluded 1`] = `
<body>
  
    
  <main
    aria-hidden="true"
  >
    
      
    <button>
      
        test
      
    </button>
    
    
  </main>
  
    
  <div>
    
      test
    
  </div>
  
  
</body>
`;

exports[`should unhide all hidden children 1`] = `
<body>
  
    
  <main>
    
      
    <button>
      
        test
      
    </button>
    
    
  </main>
  
    
  <div>
    
      test
    
  </div>
  
  
</body>
`;

exports[`shouldn't hide when there are hidden children 1`] = `
<body>
  
    
  <main
    aria-hidden="true"
  >
    
      
    <button>
      
        test
      
    </button>
    
    
  </main>
  
    
  <div
    aria-hidden="true"
  >
    
      test
    
  </div>
  
  
</body>
`;
